plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -129,66 +131,97 @@ public Network design(NetworkOffering offering, DeploymentPlan plan, Network use
+                Vpc vpcObj = _vpcDao.findById(vpcId);
+                vpcName = vpcObj.getName();
+                vpcUuid = vpcObj.getUuid();
+                preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspVpcDomainTemplateName.key());
+            } else {
+                if (offering.getGuestType() == GuestType.Shared) {
+                    isL3Network = true;
+                    isSharedNetwork = true;
+                    preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspSharedNetworkDomainTemplateName.key());
+
+                    ipAddressRanges = new ArrayList<String[]>();
+                    List<VlanVO> vlans = _vlanDao.listVlansByNetworkId(network.getId());
+                    for (VlanVO vlan : vlans) {
+                        boolean isIpv4 = StringUtils.isNotBlank(vlan.getIpRange());
+                        String[] range = isIpv4 ? vlan.getIpRange().split("-") : vlan.getIp6Range().split("-");
+                        ipAddressRanges.add(range);
+                    }
+                } else if (isL3Network(network)) {
+                    isL3Network = true;
+                    preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspIsolatedNetworkDomainTemplateName.key());
+                }
+            }
+
+            HostVO nuageVspHost = getNuageVspHost(physicalNetworkId);
+            List<String> dnsServers = _nuageVspManager.getDnsDetails(network);
+            List<String> gatewaySystemIds = _nuageVspManager.getGatewaySystemIds();
+            ImplementNetworkVspCommand cmd = new ImplementNetworkVspCommand(networksDomain.getName(), networksDomain.getPath(), networksDomain.getUuid(), networksAccount.getAccountName(),
+                    networksAccount.getUuid(), network.getName(), network.getCidr(), network.getGateway(), network.getNetworkACLId(), dnsServers, gatewaySystemIds, network.getUuid(),
+                    isL3Network, isVpc, isSharedNetwork, vpcName, vpcUuid, offering.getEgressDefaultPolicy(), ipAddressRanges, preConfiguredDomainTemplateName);
+            ImplementNetworkVspAnswer answer = (ImplementNetworkVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmd);
 
-        if (answer == null || !answer.getResult()) {
-            s_logger.error("ImplementNetworkNuageVspCommand failed");
-            if ((null != answer) && (null != answer.getDetails())) {
-                s_logger.error(answer.getDetails());
+            if (answer == null || !answer.getResult()) {
+                s_logger.error("ImplementNetworkNuageVspCommand failed");
+                if ((null != answer) && (null != answer.getDetails())) {
+                    s_logger.error(answer.getDetails());
+                }
+                return null;
             }
-            return null;
+            s_logger.info("Implemented OK, network " + networkUuid + " in tenant " + tenantId + " linked to " + implemented.getBroadcastUri().toString());
```

Comment:
The call to ``implemented.getBroadcastUri().toString()`` in extraneous.  Append ``implemented.getBroadcastUri()`` and the compiler will automatically insert the ``toString()`` call.